doc/sphinx-guides/source/admin/metadatacustomization.rst

```diff
@@ -593,10 +593,32 @@ Running a curl command like "load" example above should make the new custom meta
 Updating the Solr Schema
 ~~~~~~~~~~~~~~~~~~~~~~~~
 
-Once you have enabled a new metadata block you should be able to see the new fields in the GUI but before you can save the dataset, you must add additional fields to your Solr schema. You should run the following curl command to have Dataverse output the "field name" and "copyField" elements for all the metadata fields that have been loaded into Dataverse:
+Once you have enabled a new metadata block you should be able to see the new fields in the GUI but before you can save
+the dataset, you must add additional fields to your Solr schema.
+
+An API endpoint of Dataverse provides you with a generated set of all fields that need to be added to the Solr schema
+configuration, including any enabled metadata schemas:
 
 ``curl http://localhost:8080/api/admin/index/solr/schema``
 
+For convenience and automation you can download :download:`updateSchemaMDB.sh <../../../../conf/solr/7.3.1/updateSchemaMDB.sh>`. It downloads, parses and writes the schema
+files before triggering a Solr reload. **You should run this as the Solr user, not root.**
+
+By default, it will download from Dataverse at `http://localhost:8080` and reload Solr at `http://localhost:8983`.
+You may use the following environment variables with this script or mix'n'match with options:
+
```

Comment:
I would maybe add that, if Solr and Dataverse are running on two different servers, the script must be run on the former - i.e., the Solr system, because it needs access to the local disk, in order to save the schema. (Thinking about it, what's the point of the -s parameter? - shouldn't the Solr ULR be *hard-coded* to `http://localhost:8983`? - because if it is a remote server, then it's likely not going to work... unless TARGET directory is on an NFS mount, or something similar, accessible to both systems... seems easier, to just say that the script *must* be run on the Solr server, and assume that it should be reachable at the localhost address, no?)

Reply:
@landreev good point and we can let @poikilotherm tell us but I'm going to guess he's thinking ahead to someday using the Solr API to load the field rather than futzing with files on disk. That's the vision of #5989, to use the Solr APIs. @poikilotherm does use the Solr API for one operation... reloading of Solr:

echo "Triggering Solr RELOAD at ${SOLR_URL}/solr/admin/cores?action=RELOAD&core=collection1"
curl -f -sS "${SOLR_URL}/solr/admin/cores?action=RELOAD&core=collection1" 

So maybe he's just trying to help guide us forward.

Reply:
@pdurbin I saw that the script was using the API for restarting Solr. Is it actually possible to use Solr API to modify its schema remotely? - I thought it wasn't. 
Regardless, there's no harm in keeping SOLR_URL configurable. But the script in its current form relies on writing the schema file, so I feel the guide should mention that it needs to be run locally.

Reply:
@landreev yes, is is possible to modify Solr's schema remotely and @pkiraly is investigating this in #5989 

While chatting with @kcondon just now it occurred to me that @poikilotherm may be more in need of configuring the **port** that Solr is running on for his script rather than the hostname. He's running Dataverse in Kubernetes so who knows what crazy stuff he's doing in there. 😄 Anyway, that might be the reason why he doesn't want to hard code a hostname and port. Just a theory.

I'll go add a line in the guide to indicate that the script must be run on the Solr server itself.

Reply:
Ok, I emphasized that the script must be run on the Solr server itself in 7b2becd9c

Reply:
Hey guys,

indeed I need hostname and port fully configurable due to usage on Kubernetes. Most likely one will provide Some K8s service IP and port. (It's also a good practice not hardcoding such things anyway...)

Please be aware that writing the files on K8s is most likely done by some sidecar container/job, so this most certainly will not run in the same container on K8s.

As the guides are mostly for people using the classic installation I am totally happy with any additions you provide. Just go ahead and push it. :smile:

Cheers!